linkerd/core/src/main/scala/io/buoyant/linkerd/RequestActiveTracer.scala

```diff
@@ -0,0 +1,265 @@
+        }.find(!_.isEmpty).toList.flatten
+      case _ => List.empty
+    }
+  }
+
+  private def getRequestTraceResponse(resp: Response, stopwatch: Stopwatch.Elapsed) = {
+
+    val EmptyDelegateTree = DelegateTree.Empty(Path.empty, Dentry.nop)
+
+    val serviceName = DstPathCtx.current match {
+      case Some(dstPath) => dstPath.path
+      case None => Path.empty
+    }
+
+    val selectedEndpoint = endpoint.addr match {
+      case inetAddr: Address.Inet => inetAddr.addr.toString.stripPrefix("/")
+      case _ => ""
+    }
+
+    val clientPath = DstBoundCtx.current match {
+      case None => Path.empty
+      case Some(addrSet) => addrSet.name.id match {
+        case p: Path => p
+        case _ => Path.empty
+      }
+    }
+
+    val lbSet = DstBoundCtx.current match {
+      case None => Future.value(Addr.Neg)
+      case Some(addrSet) => addrSet.name.addr.changes.toFuture
+    }
+
+    val addresses = lbSet.map {
+      case Addr.Bound(a, _) => Some(
+        a.map {
+          case inetAddr: Address.Inet => inetAddr.addr.toString.stripPrefix("/")
+          case _ => ""
+        }
+      )
+      case _ => None
+    }
+
+    val dtreeF = namers.interpreter match {
+      case delegator: Delegator => delegator.delegate(dtab.dtab(), serviceName)
+        .map(Some(_))
+      case _ => Future.None
+    }
+
+    dtreeF.joinWith(addresses) {
+      case (dTree: Option[DelegateTree[Name.Bound]], addrSet: Option[Set[String]]) =>
+        val tree = formatDelegation(dTree.getOrElse(EmptyDelegateTree), List.empty, clientPath)
+
+        resp.contentString = formatRouterContext(
+          resp.contentString,
+          serviceName.show,
+          clientPath.show,
+          selectedEndpoint,
+          addrSet,
+          tree.map(_.toString),
+          stopwatch().inMillis.toString
+        )
+
+        //We set the content length of the response in order to view the content string entirely
+        resp.contentLength = resp.content.length
+        resp
+    }
+  }
+
+  override def apply(
+    req: Request,
+    svc: Service[Request, Response]
+  ): Future[Response] = {
+
+    val maxTraceDepth = Try(req.headerMap.get(MaxForwardsHeader).map(_.toInt))
+    val addRouterCtxt = req.headerMap.get(AddRouterContextHeader)
+
+    (req.method, maxTraceDepth, addRouterCtxt) match {
+      case (Method.Trace, Throw(_: NumberFormatException), _) =>
+        // Max-Forwards header is unparseable
+        val resp = Response(Status.BadRequest)
+        resp.contentString = s"Invalid value for $MaxForwardsHeader header"
+        Future.value(resp)
+      case (Method.Trace, Return(Some(0)), isAddRouterCtxt) =>
+        // Max-Forwards header is 0 and the l5d-add-context header is present
+        isAddRouterCtxt match {
+          // returns a response with no router context
+          case None => Future.value(Response(req))
+
+          // returns a response with router context
+          case _ =>
```

Comment:
We probably want to be more strict than allowing any value to be treated as `true`.  `com.twitter.finagle.util.StringUtil.toBoolean` seems like a good thing to use here.  It will map "1" and "t" and "true" to true and everything else to false.